Homework_3

mariohuq opened the pull request:
- [x] BEM
- [x] Переключение темы

Commit:
Fix warnings

Commit:
Change classes to match BEM, combine margins into one rule, remove unused classes, replace download form by download link

Commit:
Introduce variable colors and icons, make light theme version, change icon implementation

Commit:
Make working theme switcher

Commit:
Center card

Leikam commented on index.css:
```diff
@@ -65,71 +86,117 @@ html, body {
-            font-weight: 500;
-            font-size: 14px;
-            line-height: 19px;
+        .minor-info .download-button {
```
все правильно: отдельно стили .download-button и отдельно дотюнил под место в .minor-info .download-button.

Leikam commented on index.css:
```diff
@@ -65,71 +86,117 @@ html, body {
 
-            margin-top: auto;
-            margin-bottom: 13px;
+        a.download-button {
```
можно, например, завести абстарктный компонент ссылки или кнопки, на нем обнулить подчеркивание, задать цвет, предусмотреть разные цвета ссылок (активная, неактивная, яркая). Тогда тут был бы такой симбио
.link.link_type-default.download-button {} А на деле

.link {
  text-decoration: none;
}

.link_type-default {
  color: var(--foreground);
}

и для самого download-button стилей бы уже и не осталось.

Leikam commented on index.css:
```diff
@@ -65,71 +86,117 @@ html, body {
+}
+
+body.light-theme {
+    --switch-icon: "☾︎";
```
хорошо :)

Leikam commented on index.html:
```diff
@@ -10,29 +10,33 @@
+        <h2 class="minor-info__name">Kondraev Dmitry</h2>
+        <p class="minor-info__position">Web Developer</p>
+        <ul class="social">
+            <li><a class="social__icon github" href="https://github.com/mariohuq"></a></li>
```
github, tamtam лучше модификаторами, чем отдельными блоками

Leikam commented on index.html:
```diff
@@ -10,29 +10,33 @@
+    </section>
+</div>
+<script>
+    document.querySelector('.theme-switcher').addEventListener('click', function() {
```
просто и хорошо

Leikam reviewed (approved):
8/10

Не хватило глубины компонентам, модификаторов, хорошо бы разбить компоненты и их стили по файлам. В нашем случае пока только стили, в дальнейшем это может быть и js и шаблоны для компонента.
А вообще хорошая работа.

Commit:
Fix https://github.com/ok-technopolis-2021/Kondraev_Dmitry/pull/3#discussion_r747859789

Commit:
Fix https://github.com/ok-technopolis-2021/Kondraev_Dmitry/pull/3#discussion_r747859345

Remove cursor: pointer on button,
Use mixes instead of group selectors,
Rename modifiers,
Use elements instead of nested selectors

Leikam reviewed (commented):
Поправил оценку на 9
